lib/grandpa/vote_message.go

```diff
@@ -234,24 +235,94 @@ func (s *Service) checkForEquivocation(voter *Voter, vote *SignedVote, stage Sub
 
 	_, has := eq[v]
 	if has {
+		// TODO check with team that I don't have to do anything here
 		// if the voter has already equivocated, every vote in that round is an equivocatory vote
 		eq[v] = append(eq[v], vote)
-		return true
+		return fmt.Errorf("%w: voter already equivocated", ErrEquivocation)
 	}
 
 	existingVote, has := s.loadVote(v, stage)
 	if !has {
-		return false
+		return nil
 	}
 
 	if has && existingVote.Vote.Hash != vote.Vote.Hash {
 		// the voter has already voted, all their votes are now equivocatory
 		eq[v] = []*SignedVote{existingVote, vote}
 		s.deleteVote(v, stage)
-		return true
+
+		err := s.reportEquivocation(stage, existingVote, vote)
+		if err != nil {
+			// TODO get feedback on if this is a appropriate way to handle error
+			logger.Errorf("%s: %s", errReportingEquivocation, err)
+			return fmt.Errorf("%w: %s", ErrEquivocation, err)
+		}
+		return fmt.Errorf("%w", ErrEquivocation)
```

Comment:
I think it's good to log as an error the equivocation report error, but it doesn't feel right to put it as context with the returned error. On the other hand, could we add some context about the equivocation (on line `258` and line `260`, which can return the same wrapped error and context)?

Maybe something like

```suggestion
		if err != nil {
			// TODO get feedback on if this is a appropriate way to handle error
			logger.Errorf("failed reporting equivocation: %s", err)
		}
		return fmt.Errorf("%w: voter %s has existing vote %s and new vote %s", 
			ErrEquivocation, v, existingVote.Vote.Hash, vote.Vote.Hash)
```

you can also remove `errReportingEquivocation`